File: lib/aws/ec2/vpcGatewayAttachment.ts
// Copyright 2016 Pulumi, Inc. All rights reserved.

import {InternetGateway} from './internetGateway';
import {VPC} from './vpc';
import * as cloudformation from '../cloudformation';

// Attaches a gateway to a VPC.
// @name: aws/ec2/vpcGatewayAttachment
// @website: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc-gateway-attachment.html
export class VPCGatewayAttachment extends cloudformation.Resource {
    constructor(name: string, args: VPCGatewayAttachmentArgs) {
        super({
            name: name,
            resource: "AWS::EC2::VPCGatewayAttachment",
            properties: args,
        });
    }
}

export interface VPCGatewayAttachmentArgs {
    // The VPC to associate with this gateway.
    readonly vpc: VPC;
    // The Internet gateway to attach to the VPC.
    readonly internetGateway: InternetGateway;
}

File: lib/aws/idl/ec2/vpcGatewayAttachment.go
// Copyright 2016-2017, Pulumi Corporation.  All rights reserved.

package ec2

import (
	"github.com/pulumi/lumi/pkg/resource/idl"
)

// VPCGatewayAttachment attaches a gateway to a VPC.  For more information, see
// http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc-gateway-attachment.html.
type VPCGatewayAttachment struct {
	idl.NamedResource
	// The VPC to associate with this gateway.
	VPC *VPC `lumi:"vpc,replaces"`
	// The Internet gateway to attach to the VPC.
	InternetGateway *InternetGateway `lumi:"internetGateway,replaces"`
}
